### PR TITLE
Chore: Apply Emitter in simulator and fix cleanup when terminated, Refactor naming

### PR DESCRIPTION
**Pull request by cizl**

### CODE REVIEW

File: src/simulator/shared.ts [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
 export type ISimulationEdge = SimulationLinkDatum<ISimulationNode> & { id: number };
 
-export interface ISimulator {
+export enum SimulatorEventType {
+  STABILIZATION_STARTED = 'stabilizationStarted',
+  STABILIZATION_PROGRESS = 'stabilizationProgress',
+  STABILIZATION_ENDED = 'stabilizationEnded',
+  NODE_DRAGGED = 'nodeDragged',
+  NODE_DRAG_ENDED = 'nodeDragEnded',
+  SETTINGS_UPDATED = 'settingsUpdated',
```
Comment (tonilastre) [inline src/simulator/shared.ts:15]: Nitpicking, but now when you cleaned up Events in simulator engines and main classes, maybe it makes sense to make naming consistent with the rest of the library where:

* Event values are present simple (e.g. `render-start`, `node-hover`, `mouse-move`)
* Event values are not camel case (e.g. `render-start`, ...)

**Review by tonilastre — APPROVED** [review]
Great refactoring! Now all events are structured as "event emitters" which is awesome!

Btw there is just one comment regarding name consistency.

File: src/simulator/shared.ts (hunk 1 above)
Comment (cizl) [reply src/simulator/shared.ts:15]: Yep! Makes sense. I'll also refactor these messages in the screenshot, as they are not written in `MACRO_CASE`.
Screenshot from 2022-10-07 12-59-49

or maybe the best example is this one where both are clearly visible:
Screenshot from 2022-10-07 13-03-15

Btw regarding naming, I have a few more questions:
* We are using both `stabilization` and `simulation`. Should we merge those two?
* In the picture above, should all of them be present simple?

File: src/views/default-view.ts [hunk 2]
```diff
@@ -27,6 +29,7 @@ export interface IDefaultViewSettings<N extends INodeBase, E extends IEdgeBase>
   isOutOfBoundsDragEnabled: boolean;
   areCoordinatesRounded: boolean;
   isSimulationAnimated: boolean;
+  areCollapsedContainerDimensionsAllowed: boolean;
```
Comment (tonilastre) [inline src/views/default-view.ts:32]: Just don't forget to add this to the `docs` for both views.

**Review by tonilastre — APPROVED** [review]

**Comment by cizl** [comment]
@tonilastre Applied the standardized naming convention and updated docs. Feel free to take one final glance. :slightly_smiling_face:

File: docs/view-map.md [hunk 3]
```diff
@@ -143,9 +156,9 @@ const defaultSettings = {
   },
   map: {
     zoomLevel: 2, // Default map zoom level
-    tile: new L.TileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png') // OpenStreetMaps
-  }
-}
+    tile: new L.TileLayer("https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"), // OpenStreetMaps
+  },
+};
```
Comment (tonilastre) [inline docs/view-map.md:161]: Add `areCollapsedContainerDimensionsAllowed` here and to the interface above.

File: src/simulator/shared.ts [hunk 4]
```diff
@@ -48,8 +67,8 @@ export interface ISimulatorEventSettings {
 export interface ISimulatorEvents {
   onNodeDrag: (data: ISimulatorEventGraph) => void;
   onNodeDragEnd: (data: ISimulatorEventGraph) => void;
-  onStabilizationStart: () => void;
-  onStabilizationProgress: (data: ISimulatorEventGraph & ISimulatorEventProgress) => void;
-  onStabilizationEnd: (data: ISimulatorEventGraph) => void;
+  onSimulationStart: () => void;
+  onSIMULATION_PROGRESS: (data: ISimulatorEventGraph & ISimulatorEventProgress) => void;
```
Comment (tonilastre) [inline src/simulator/shared.ts:71]: Khm :D

File: src/views/default-view.ts [hunk 5]
```diff
@@ -49,7 +50,7 @@ export class DefaultView<N extends INodeBase, E extends IEdgeBase> implements IO
 
   private _isSimulating = false;
   private _onSimulationEnd: (() => void) | undefined;
-  private _simulationStartedAt = Date.now();
+  private _simulationStartedAtAt = Date.now();
```
Comment (tonilastre) [inline src/views/default-view.ts:53]: Double `At`.

**Review by tonilastre — CHANGES_REQUESTED** [review]
Some minor fixes are needed.

File: src/simulator/shared.ts (hunk 4 above)
Comment (cizl) [reply src/simulator/shared.ts:71]: It was bound to happen :sweat_smile:

File: docs/view-default.md [hunk 6]
```diff
@@ -204,7 +206,7 @@ that allows Orb to position the nodes.
 ![](./assets/view-default-fixed.png)
 
 ```typescript
-import { DefaultView } from '@memgraph/orb';
+import { DefaultView } from "@memgraph/orb";
```
Comment (cizl) [inline docs/view-default.md:209]: We have to align our formatters at some point. Seems like every time we open a markdown file and save it we have different outputs.

**Review by tonilastre — APPROVED** [review]

**Pull request merged**